### DISCUSSION TITLE
hit marker shows up too early in editor

### DISCUSSION THREAD

**Discussion by papertek** [ppy/osu#22726] [Q&A] [answered]
dunno if this is intentional but here's a comparison -

Stable:
![image](https://user-images.githubusercontent.com/72106503/220905850-e3455ac8-9e71-4f49-bf54-f5b5f6672280.png)

Lazer:
![image](https://user-images.githubusercontent.com/72106503/220905779-4a9d69e1-5ae0-4285-8e3a-408e43ebe4b0.png)

Video:
https://user-images.githubusercontent.com/72106503/223320085-dd0c6d0e-227a-4dbe-b657-e4f63abbacf5.mp4

**Comment by Axilotl17** [comment]
please include map and timestamp (you should also be submitting logs as well). this CAN be intended behavior: because slider ends actually get hit slightly early to make very fast sliders bearable, and this is intentional. however, if this is a slow slider, then it shouldn't be happening. 

![image](https://user-images.githubusercontent.com/78441708/221383653-5106e96d-8fbe-472d-958d-24109d52183f.png)
06:32:850 (1)
https://osu.ppy.sh/beatmapsets/1651410#osu/3371379

this instance is INTENTIONAL. 


the "white circle" is called a "hit marker", by the way.

**Reply by papertek** [reply]
ok

**Reply by Axilotl17** [reply]
>please include map and timestamp

???

**Reply by papertek** [reply]
this map isnt uploaded, but heres the same issue in [another map](https://osu.ppy.sh/beatmapsets/1318077#osu/2730751)
05:25:876 (2) - slider

in comparison to stable, "it feels weird"(tm)

**Reply by Axilotl17** [reply]
![image](https://user-images.githubusercontent.com/78441708/222639850-9b909ea6-827e-4abf-8426-8ff9d05c5a00.png)
if you mean this, this is intentional. stable has this too, its just not visible in the editor the same way it is on lazer. 
you can also turn off the hit markers by clicking "view" in the top of the editor, and this will make it more like stable

**Reply by papertek** [reply]
> its just not visible in the editor the same way it is on lazer.

this is what I'm reporting, the slider end hitmaker appearing too early.
I'm reporting this because you would expect the circle to be *hit* (and play a hitsound) because that's how the behavior is in stable.

**Reply by Axilotl17** [reply]
well, this is intentional to show how the end of the slider is actually hit. in both lazer and stable, when you PLAY a map, the slider end is actually hit before you reach the end of the slider, to make fast sliders more playable. as you've seen here, this behavior is replicated in the editor.

stable doesnt have this behavior at all, because stable doesn't have hit markers at all. if you want the stable experience, TURN OFF hit markers.

**Reply by papertek** [reply]
that isnt the point of this issue and i think you're misunderstanding it. i *dont* want the hit marker gone. i already gave the example behavior in the screenshots above, I'll rather wait for an official answer

**Reply by Axilotl17** [reply]
to clarify, your issue is that the hit marker should display when the slider actually ends on beat, not when it ends in gameplay?

if so, I see no relation to stable behavior.

**Comment by peppy** [comment, marked as answer]
Will track here: https://github.com/ppy/osu/issues/22805
